### PR TITLE
MAP-13: MetadataTerm.metadataClass and MetadataTerm.metadataUuid shou…

### PR DESCRIPTION
**Pull request by jvena1**
ref: https://issues.openmrs.org/browse/MAP-13

### CODE REVIEW

File: omod/src/main/resources/liquibase.xml [hunk 1]
```diff
@@ -195,5 +195,31 @@
 							 tableName="metadatamapping_metadata_source"
 							 columnNames="name" />
 	</changeSet>
-
+	
+	<changeSet id="metadatamapping-2016-01-06-0800" author="jasonvena">
+		<preConditions onFail="MARK_RAN">
+			<columnExists tableName="metadatamapping_metadata_term_mapping"
+						  columnName="metadata_class" />
+		</preConditions>
```
Comment (suniala) [inline omod/src/main/resources/liquibase.xml]: What this precondition means is: if the column metadata_class does not exist, mark this changeset as having been already ran and don't do anything else. Perhaps you meant to check if the constraint exists? Btw, I'm personally not sure if this kind of preconditions are useful or not but they seem to be a standard practice in OpenMRS and I guess it's a good idea to follow the established practices.

http://www.liquibase.org/documentation/preconditions.html

File: omod/src/main/resources/liquibase.xml [hunk 2]
```diff
@@ -195,5 +195,31 @@
...
+		<dropNotNullConstraint tableName="metadatamapping_metadata_term_mapping"
+							   columnName="metadata_class"
+							   columnDataType="varchar(1024)" />
+	</changeSet>
+	
+	<changeSet id="metadatamapping-2016-01-06-0801" author="jasonvena">
+		<preConditions onFail="MARK_RAN">
+			<columnExists tableName="metadatamapping_metadata_term_mapping"
+						  columnName="metadata_uuid" />
+		</preConditions>
```
Comment (suniala) [inline omod/src/main/resources/liquibase.xml]: Ditto.

File: omod/src/main/resources/liquibase.xml (hunk 1 above)
Comment (jvena1) [inline omod/src/main/resources/liquibase.xml]: Hi Mikko, I have intended it as written, following examples from core developers such as seen [here](https://github.com/openmrs/openmrs-core/blob/cdbf9421212f5f5d48fa9d92c42515e4c517f20c/api/src/main/resources/liquibase-update-to-latest.xml#L7696) and [here](https://github.com/openmrs/openmrs-core/blob/cdbf9421212f5f5d48fa9d92c42515e4c517f20c/api/src/main/resources/liquibase-update-to-latest.xml#L3149).

File: omod/src/main/resources/liquibase.xml (hunk 2 above)
Comment (jvena1) [inline omod/src/main/resources/liquibase.xml]: Same response as prior.

File: omod/src/main/resources/liquibase.xml (hunk 1 above)
Comment (suniala) [inline omod/src/main/resources/liquibase.xml]: Well, I said it myself: it's a good idea to follow established practices. I appreciate the fact that you have checked out what others have done. However, in this case I find the established practice a bit problematic. Let's see what others think: https://talk.openmrs.org/t/liquibase-precondition-conventions/4282

**Pull request closed without merging**